universe/base.go

```diff
@@ -298,6 +311,58 @@ func (a *MintingArchive) RegisterNewIssuanceBatch(ctx context.Context,
 	return nil
 }
 
+// getPrevAssetSnapshot returns the previous asset snapshot for the passed
+// proof. If the proof is a genesis proof, then nil is returned.
+func (a *MintingArchive) getPrevAssetSnapshot(ctx context.Context,
+	uniID Identifier, newProof proof.Proof) (*proof.AssetSnapshot, error) {
+
+	// If this is a genesis proof, then there is no previous asset (and
+	// therefore no previous asset snapshot).
+	if newProof.Asset.HasGenesisWitness() {
+		return nil, nil
+	}
+
+	// Query for proof associated with the previous asset.
+	prevID, err := newProof.Asset.PrimaryPrevID()
```

Comment:
We're still missing a `nil` check for the `prevID`. Normally we'd validate a proof first so this wouldn't be necessary. But because we're accessing fields from it before we've validated it, someone could craft a malicious proof and get the universe to panic when we access `prevID.ScriptKey` below.

Reply:
I've added a `prevID == nil` check below this function call.